src/app/services/task.service.ts

```diff
@@ -12,15 +13,33 @@ export class TaskService {
     this.tasksArray.unshift(task);
   }
 
-  deleteTask(deleteTaskId: number){
-    const indexOfTask = this.tasksArray.findIndex(task => {
-      return task.id === deleteTaskId;
-    });
-    
+  deleteTask(deleteTaskId: number) {
+    const indexOfTask = this.getTaskIndex(deleteTaskId);
     this.tasksArray.splice(indexOfTask,1);
   }
 
+  updateTaskStatus(completedTaskId: number) {
+    const indexOfTask = this.getTaskIndex(completedTaskId);
+    const daysToComplete = this.getDaysToComplete(indexOfTask)
+    this.tasksArray[indexOfTask].status = true;
+    this.tasksArray[indexOfTask].daysToComplete = daysToComplete;
+  }
+
   getTasks() {
     return this.tasksArray;
   }
+
+  getTaskIndex(id: number) {
+    const index = this.tasksArray.findIndex(task => {
+      return task.id === id;
+    });
+    return index;
+  }
+
+  getDaysToComplete(index: number) {
+    const doneDate = formatDate (new Date(), 'dd.MM.YY', 'en')
+    const createDate = this.tasksArray[index].createDate
+    const diffTime = Math.abs(parseFloat(doneDate)  - parseFloat(createDate));
```

Comment:
Even if the user takes only a few minutes, it should still be counted as one day. change the logic